### PR TITLE
Updated Unitary calculator following new code generation style.

### PR DESCRIPTION
**Pull request by MichaelBroughton**
Updates unitary calculator to use the same style of code in the rest of the library. Hopefully when #227 is resolved, support for unitary calculator can be included more long term :). This is the first step in resolving #263 .

### CODE REVIEW

File: lib/unitary_calculator_basic.h [hunk 1]
```diff
@@ -197,209 +161,76 @@ class UnitaryCalculatorBasic final {
...
+                      unsigned sqrt_size, fp_type* rstate) {
+      fp_type rn, in;
+      fp_type rs[16], is[16];
+
+      auto row_size = uint64_t{1} << n_qb;
+      uint64_t i = ii % sqrt_size;
+      uint64_t j = ii / sqrt_size;
+
+      uint64_t k = (1 * i & ms[0]) | (2 * i & ms[1]) | (4 * i & ms[2]);
+      uint64_t kk = (1 * j & ms[0]) | (2 * j & ms[1]) | (4 * j & ms[2]);
```
Comment (95-martin-orion) [inline lib/unitary_calculator_basic.h:197]: Same here - rename to avoid collision with `k` used in loop.

**Review by 95-martin-orion — APPROVED** [review]
Looks good modulo some variable names. Is my understanding correct that the changes between `simulator_basic` and this code are due to manipulating a vector vs. the unitary matrix?

File: lib/unitary_calculator_basic.h [hunk 2]
```diff
@@ -121,68 +80,73 @@ class UnitaryCalculatorBasic final {
...
+      fp_type rn, in;
+      fp_type rs[4], is[4];
+
+      auto row_size = uint64_t{1} << n_qb;
+
+      uint64_t i = ii % sqrt_size;
+      uint64_t j = ii / sqrt_size;
+
+      uint64_t k = (1 * i & ms[0]) | (2 * i & ms[1]);
+      uint64_t kk = (1 * j & ms[0]) | (2 * j & ms[1]);
```
[the remark this answers is not in the document]
Comment (MichaelBroughton) [reply lib/unitary_calculator_basic.h]: Fixed.

File: lib/unitary_calculator_basic.h (hunk 1 above)
Comment (MichaelBroughton) [reply lib/unitary_calculator_basic.h]: Done.

**Comment by MichaelBroughton** [comment]
> Is my understanding correct that the changes between simulator_basic and this code are due to manipulating a vector vs. the unitary matrix?

Yes. The unitary calculator is just like the state vector simulators with two major changes:
1. Larger underlying array (grows like 2**n * 2**n instead of 2**n)
2. The indexing routines and gate application logic have to account for the row and column when applying the gate instead of just the one long row (or column) that is done in the state vector simulator.

**Pull request merged**